pkg/process/cache.go

```diff
@@ -121,8 +121,9 @@ func (pc *Cache) refInc(p *ProcessInternal) {
 	atomic.AddUint32(&p.refcnt, 1)
 }
 
-func (pc *Cache) Purge() {
+func (pc *Cache) purge() {
```

Comment:
there's another Purge instance in here.. https://github.com/cilium/tetragon/blob/95bd82351ad8b39245f1feaf6c72ce2b4a3fcd99/pkg/process/cache.go#L49  but not sure it's needed

Reply:
The flow is: `purge()` sends to `stopChan`, then `cacheGarbageCollector()` reads from this channel and actually purges the inner `lru.Cache` (the line you pointed). So metric reset happens in a different goroutine than actual cleanup, but it should be fine.